ProcessQueue/__init__.py

```diff
@@ -476,3 +479,71 @@ def main(msg: func.QueueMessage):
             field_ids = []
             field_names = []
             data = []
+    if parent_sr_id is not "None":
+        # Grab Parent Scan Report
+        parent_scan_report = requests.get(
+                    url=api_url
+                    + "scanreporttablesfilter/?scan_report="
+                    + str(parent_sr_id),
+                    headers=headers,
+                )
+        parent_table_ids=[]
+        parent_tables=[]
+        child_tables=[]
+        
+        child_scan_report=requests.get(
+                    url=api_url
+                    + "scanreporttablesfilter/?scan_report="
+                    + str(scan_report_id),
+                    headers=headers,
+                )
+        
+        response_parent = json.loads(parent_scan_report.content.decode("utf-8"))
+        response_child=json.loads(child_scan_report.content.decode("utf-8"))
+
+        print("PARENT SCAN REPORT TABLES:",response_parent)
+        print("CHILD SCAN REPORT TABLES",response_child)
+
+        # Find differences between JSON?
+        # difference=diff(response_parent, response_child)
+        # print("JSON DIFF",difference)
+
+        # Find differences of two lists?
+        for element in range(len(response_parent)):
+            parent_tables.append(response_parent[element]["name"])
+            parent_table_ids.append(response_parent[element]["id"])
+        for element in range(len(response_child)):
+            child_tables.append(response_child[element]["name"])
+        # Check if tables are the same
+        if parent_tables==child_tables:

```

Comment:
is this safe? i dont know if the responses are ordered - by that i mean, if the elements in the two lists are definitely going to be in the same order.

`[1,2,3,4] == [1,2,4,3]` : `False` when you might want this to be `True` as the lists are different but the elements are the same?

`sorted(parent_tables) == sorted(child_tables)` would be the solution to that (?)

Reply:
I think the wider question is whether this is the right thing to be doing. We have a special case where `sorted(parent_tables) == sorted(child_tables)` but I don't know that it's all that useful to pick it out. Ultimately, whether or not we have absolutely the same or not, we only want the intersection of tables and intersection of fields within tables (and values within fields). So I don't think we need to spend time and code on the very special case where they're identical in every respect - that will be rare. Instead, just trim down to the intersection of the 2 scan reports (so where the fields in tables are the same, and where the values inside those fields are the same) and process those. Whether that's the whole table or not doesn't really matter.

Reply:
in sum (sorry for the abuse of all sorts of notations) :
```
for table in intersection(parent_tables, child_tables):
    for field in intersection(parent_tables[table].fields, child_tables[table].fields):
        copy field concepts
        for value in intersection(parent_tables[table].fields[field].values, child_tables[table].fields[field].values):
            copy value concepts
```